### PR TITLE
Accessing Park API, refactoring "gets", refactoring food and attractions

### PR DESCRIPTION
**Pull request by ThomasELong**
We built...
- a means to access the NPS full park API
- a means to access parks based off of chosen State
- a means to pull food and attractions based on State, rather than park
- the weather provision to be based on chosen State, otherwise the user is routed to weather site

-Verify no extra code is written
- Verify full functionality

### CODE REVIEW

**Review by alexcurnow — APPROVED** [review]
👍

**Review by durrjp — APPROVED** [review]
APPROVED

**Review by heidijane — APPROVED** [review]
👍

File: scripts/attractions/AttractionSelectComponent.js [hunk 1]
```diff
@@ -41,12 +43,20 @@ eventHub.addEventListener("parkDropDownChanged", event => {
...
-    const filteredAttractions = attractions.filter(attraction => attraction.state === chosenParkState)
+    const chosenParkStates = chosenParkObject.states
+     // creating a workable array of states from the string of states
+     const chosenParkStatesArray = chosenParkStates.split(",")
+    
+     // iterate over array of states (within a park) and add each attraction within those states to filtered attractions array 
+     let filteredAttractions = []
+     for (const state of chosenParkStatesArray) {
+         const foundAttractions = attractions.filter(attraction => attraction.state === state)
+         Array.prototype.push.apply(filteredAttractions,foundAttractions)
```
Comment (stevebrownlee) [inline scripts/attractions/AttractionSelectComponent.js:54]: Is this cleverness for clevernesses sake?

File: scripts/foods/FoodSelectComponent.js [hunk 2]
```diff
@@ -28,23 +28,35 @@ const render = (filteredFoods) => {
...
+    // pull states property from chosen park object in the form of a string
+    const chosenParkStates = chosenParkObject.states
+    // creating a workable array of states from the string of states
+    const chosenParkStatesArray = chosenParkStates.split(",")
+    
+    // iterate over array of states (within a park) and add each restaurant within those states to filtered foods array 
+    let filteredFoods = []
+    for (const state of chosenParkStatesArray) {
+        const foundFood = foods.filter(food => food.state === state)
+        Array.prototype.push.apply(filteredFoods,foundFood)
```
Comment (stevebrownlee) [inline scripts/foods/FoodSelectComponent.js:54]: Why this?

**Pull request merged**